crates/solvers/src/util/serialize/hex.rs

```diff
@@ -0,0 +1,54 @@
+use serde::{de, Deserialize, Deserializer, Serializer};
+use serde_with::{DeserializeAs, SerializeAs};
+use std::borrow::Cow;
+
+/// Serialize and deserialize binary data as a hexadecimal string.
+#[derive(Debug)]
+pub struct Hex;
+
+impl<'de> DeserializeAs<'de, Vec<u8>> for Hex {
+    fn deserialize_as<D: Deserializer<'de>>(deserializer: D) -> Result<Vec<u8>, D::Error> {
+        let s = Cow::<str>::deserialize(deserializer)?;
+        let s = s
+            .strip_prefix("0x")
+            .ok_or_else(
+                || format!("failed to decode {s:?} as a hex string: missing \"0x\" prefix",),
```

Comment:
This was needed because the compiler was unable to infer the type when returning a `de::Error::custom` directly with `ok_or_else`.